src/UserGuide/develop/QuickStart/InterfaceDefinition/InterfaceDefinition-CPP.md

```diff
@@ -0,0 +1,267 @@
+     *
+     * @return Returns 0 on success, or a non-zero error code on failure.
+     */
+    int close();
+};
+```
+
+### TableSchema
+
+Describe the data structure of the table schema
+
+```cpp
+/**
+* @brief Represents the schema information for an entire table.
+*
+* This class holds the metadata necessary to describe how a specific table is structured,
+* including its name and the schemas of all its columns.
+*/
+class TableSchema {
+    public:
+    /**
+     * Constructs a TableSchema object with the given table name, column schemas, and column categories.
+     *
+     * @param table_name The name of the table. Must be a non-empty string.
+     *                   This name is used to identify the table within the system.
+     * @param column_schemas A vector containing pointers to ColumnSchema objects.
+     *                       Each ColumnSchema defines the schema for one column in the table.
+     */
+    TableSchema(const std::string& table_name,
+                const std::vector<ColumnSchema>& column_schemas);
+};
+
+
+/**
+* @brief Represents the schema information for a single column.
+*
+* This structure holds the metadata necessary to describe how a specific column is stored,
+* including its name, data type, category.
+*/
+struct ColumnSchema {
+    std::string column_name_;
+    common::TSDataType data_type_;
+    ColumnCategory column_category_;
+
+    /**
+     * @brief Constructs a ColumnSchema object with the given parameters.
+     *
+     * @param column_name The name of the column. Must be a non-empty string.
+     *                    This name is used to identify the column within the table.
+     * @param data_type The data type of the measurement, such as INT32, DOUBLE, TEXT, etc.
+     *                  This determines how the data will be stored and interpreted.
+     * @param column_category The category of the column indicating its role or type
+     *                        within the schema, e.g., FIELD, TAG.
+     *                        Defaults to ColumnCategory::FIELD if not specified.
+     * @note It is the responsibility of the caller to ensure that `column_name` is not empty.
+     */
+    ColumnSchema(std::string column_name, common::TSDataType data_type,
+                 ColumnCategory column_category = ColumnCategory::FIELD) : column_name_(std::move(column_name)),
+                                                                           data_type_(data_type),
+                                                                           column_category_(column_category) {
+    }
+};
+
+/**
+ * @brief Represents the data type of a measurement.
+ *
+ * This enumeration defines the supported data types for measurements in the system.
+ */
+enum TSDataType : uint8_t {
+    BOOLEAN = 0,
+    INT32 = 1,
+    INT64 = 2,
+    FLOAT = 3,
+    DOUBLE = 4,
+    TEXT = 5,
+    VECTOR = 6,
+    STRING = 11,
+    NULL_TYPE = 254,
+    INVALID_DATATYPE = 255
+};
```

Comment:
Remove currently unsupported data types/encodings/compressions or those should not be used (like VECTOR) by the API user from the doc.